File: Cpp/count_good_nodes.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int arr = 0;
    int goodNodes(TreeNode* root) {
        dfs(root, root->val);
        return arr;
    }

    void dfs(TreeNode* node, int maxVal){
        if (node->val >= maxVal){
            arr++;
            maxVal = node->val;
        }
        if (node->left != NULL) dfs(node->left, maxVal);
        if (node->right != NULL) dfs(node->right, maxVal);
    }
};

File: Python/count_good_nodes.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        self.arr = 0
        def dfs(node, highest):
            if node.val >= highest:
                self.arr += 1
                highest = node.val
            if node.left:
                dfs(node.left, highest)
            if node.right:
                dfs(node.right, highest)
        dfs(root, root.val)
        return self.arr